processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -151,22 +151,28 @@ public static ColumnSelectorFactory makeColumnSelectorFactory(
           final ComplexMetricExtractor extractor = serde.getExtractor();
           return new ColumnValueSelector()
           {
+            @Override
+            public boolean isNull()
+            {
+              return in.get().getMetric(column) == null;
+            }
+
             @Override
             public long getLong()
             {
-              return in.get().getMetric(column).longValue();
+              return DimensionHandlerUtils.nullToZero(in.get().getMetric(column)).longValue();
```

Comment:
Should be similar to `RowBasedColumnSelectorFactory`.

Reply:
done.